### PR TITLE
Updated github.com/gogo/protobuf from v0.5 SHA to v1.0.0 tag

### PR DESCRIPTION
**Pull request by vllry**
This PR moves github.com/gogo/protobuf from the SHA for v0.5 to the tag for v1.0.0. It did not seem possible to use the v0.5 tag with go modules.

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Part of #79234 gomod SHA->tag cleanup.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -5,12 +5,13 @@ module k8s.io/api
...
 
 require (
-	github.com/gogo/protobuf v0.0.0-20171007142547-342cbe0a0415
+	github.com/gogo/protobuf v1.0.0
 	github.com/stretchr/testify v1.3.0
 	k8s.io/apimachinery v0.0.0
 )
 
 replace (
+	github.com/gogo/protobuf => github.com/gogo/protobuf v0.0.0-20171007142547-342cbe0a0415
```
Comment (liggitt) [inline staging/src/k8s.io/api/go.mod]: Hmmm… this is unexpected. This means we did not actually bump the dependency version. Did you use hack/pin-dependency.sh to update it to v1.0.0?

Comment (vllry) [reply staging/src/k8s.io/api/go.mod]: Whoops, sorry, I was still using gomod commands manually. I'll try that again.

**Comment by roycaihw** [comment]
Do we have tests to make sure this change is compatible with etcd?

/cc @jpbetz

**Comment by draveness** [comment]
The lint-dependencies shell script suggests using v1.1.1, could we take a look at into this?

```go
$ ./hack/lint-dependencies.sh
github.com/gogo/protobuf
    pinned:    v0.0.0-20171007142547-342cbe0a0415
    preferred: v1.1.1
    hack/pin-dependency.sh github.com/gogo/protobuf v1.1.1
```

/hold
feel free to cancel this if we decide to ignore the suggestions from lint dependencies script.

**Comment by liggitt** [comment]
this bump to v1.0.0 is very small, and it's probably better to get on a tagged version now. the prometheus bump will require more review (and I agree, if we bump that, we should update to the corresponding minimum protobuf library)

**Comment by draveness** [comment]
> this bump to v1.0.0 is very small, and it's probably better to get on a tagged version now. the prometheus bump will require more review (and I agree, if we bump that, we should update to the corresponding minimum protobuf library)

SGTM, I'll cancel the hold

/hold cancel

**Comment by liggitt** [comment]
looks like two proto generation scripts need running, and still waiting on node approval

**Comment by vllry** [comment]
/retest pull-kubernetes-dependencies

Looks like it might be a git flake.

**Comment by BenTheElder** [comment]
/retest
(/test takes a test name, /retest takes no arguments and re-runs all
failing statuses https://prow.k8s.io/command-help)

**Comment by cblecker** [comment]
@vllry Looks like it still needs a rebase

**Pull request merged**